integration_tests/src/tests/prove_commit_niporep_test.rs

```diff
@@ -0,0 +1,120 @@
+use fil_actor_miner::{ProveCommitSectorsNIParams, SectorNIActivationInfo};
+use fil_actors_runtime::test_utils::make_sealed_cid;
+use vm_api::trace::ExpectInvocation;
+use vm_api::util::apply_ok;
+use vm_api::VM;
+
+use crate::util::{
+    create_accounts, create_miner, deadline_state, override_compute_unsealed_sector_cid,
+    sector_info,
+};
+
+#[vm_test]
+pub fn prove_commit_sectors_aggregate_niporep_test(v: &dyn VM) {
+    // Expectations depend on the correct unsealed CID for empty sector.
+    override_compute_unsealed_sector_cid(v);
+    let addrs = create_accounts(v, 3, &TokenAmount::from_whole(10_000));
+    let seal_proof = RegisteredSealProof::StackedDRG32GiBV1P2_Feat_NiPoRep;
+    let (owner, worker, _, _) = (addrs[0], addrs[0], addrs[1], addrs[2]);
+    let worker_id = worker.id().unwrap();
+    let (maddr, _) = create_miner(
+        v,
+        &owner,
+        &worker,
+        seal_proof.registered_window_post_proof().unwrap(),
+        &TokenAmount::from_whole(8_000),
+    );
+    let miner_id = maddr.id().unwrap();
+    let policy = Policy::default();
+
+    // Onboard a batch of sectors
+    let expiration = v.epoch() + policy.min_sector_expiration;
+    let seal_rand_epoch = v.epoch();
+    let first_sector_number: SectorNumber = 100;
+    let manifests = vec![
+        first_sector_number,
+        first_sector_number + 1,
+        first_sector_number + 2,
+        first_sector_number + 3,
+        first_sector_number + 4,
+    ];
+    let proving_deadline = 7;
+
+    let sectors_info: Vec<SectorNIActivationInfo> = manifests
+        .iter()
+        .map(|sector_number| SectorNIActivationInfo {
+            sealing_number: *sector_number,
+            sealer_id: miner_id,
+            sector_number: *sector_number,
+            sealed_cid: make_sealed_cid(format!("sn: {}", sector_number).as_bytes()),
+            seal_rand_epoch,
+            expiration,
+        })
+        .collect();
+
+    // Prove-commit NI-PoRep
+    let aggregate_proof = RawBytes::new(vec![1, 2, 3, 4]);
+    let params = ProveCommitSectorsNIParams {
+        sectors: sectors_info.clone(),
+        seal_proof_type: RegisteredSealProof::StackedDRG32GiBV1P2_Feat_NiPoRep,
+        aggregate_proof,
+        aggregate_proof_type: RegisteredAggregateProof::SnarkPackV2,
+        proving_deadline,
+        require_activation_success: true,
+    };
+
+    let activation_epoch = v.epoch() + policy.max_prove_commit_ni_randomness_lookback / 2;
+    v.set_epoch(activation_epoch);
+
+    apply_ok(
+        v,
+        &worker,
+        &maddr,
+        &TokenAmount::zero(),
+        MinerMethod::ProveCommitSectorsNI as u64,
+        Some(params.clone()),
+    );
+
+    ExpectInvocation {
+        from: worker_id,
+        to: maddr,
+        method: MinerMethod::ProveCommitSectorsNI as u64,
+        params: Some(IpldBlock::serialize_cbor(&params).unwrap()),
+        subinvocs: None,
+        events: None,
```

Comment:
I think we need to change this .. maybe another FIP oversight; I think we need a `sector-activated` event:

After `pay_aggregate_seal_proof_fee`, for each sector, with an empty `pieces: Vec<(Cid, u64)>`:

```rust
            emit::sector_activated(rt, sector.info.sector_number, unsealed_cid, &pieces)?;
```

See PCS3 for example.